src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePluginTest.java

```diff
@@ -97,761 +97,441 @@
-		// set up
-		String imageId = FAKE_IMAGE_ID;
-		Map<String, String> imageSizeMap = null;
-		int expected = ZERO_VALUE;
-
-		// exercise
-		int value = this.plugin.getDiskSizeFromImageSizeMap(imageSizeMap, imageId);
 
-		// verify
-		Assert.assertEquals(expected, value);
-	}
-    
-	// test case: When calling the getDisckSizeFromImages method with the null
-	// imageId parameter, it must log the error that occurred and return the value
-	// zero.
+	// test case: when invoking findSmallestFlavor with valid client and compute order, the plugin
+	// should return an appropriate flavor in case one fits the order hardware requirements
 	@Test
-	public void testGetDiskSizeFromImagesWithImageIdNull() {
+	public void testFindSmallestFlavor() throws UnexpectedException, NoAvailableResourcesException {
 		// set up
-		String imageId = null;
-		Map<String, String> imageSizeMap = new HashMap<String, String>();
-		int expected = ZERO_VALUE;
+		Mockito.doReturn(this.hardwareRequirements).when(this.plugin).getBestFlavor(
+				Mockito.any(Client.class), Mockito.any(ComputeOrder.class));
 
 		// exercise
-		int value = this.plugin.getDiskSizeFromImageSizeMap(imageSizeMap, imageId);
+		this.plugin.findSmallestFlavor(this.client, this.computeOrder);
 
 		// verify
-		Assert.assertEquals(expected, value);
+		Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).getBestFlavor(
+				Mockito.eq(this.client), Mockito.eq(this.computeOrder));
 	}
-    
-	// test case: When calling the getDisckSizeFromImages method with the two null
-	// parameters, it must log the error that occurred and return the value zero.
-	@Test
-	public void testGetDiskSizeFromImagesWithTwoParametersNull() {
-		// set up
-		String imageId = null;
-		Map<String, String> imageSizeMap = null;
-		int expected = ZERO_VALUE;
-
-		// exercise
-		int value = this.plugin.getDiskSizeFromImageSizeMap(imageSizeMap, imageId);
 
-		// verify
-		Assert.assertEquals(expected, value);
-	}
-    
-	// test case: When calling the convertToInteger method with a valid numeric
-	// string, it will be converted to a valid integer value.
+	// test case: when invoking findSmallestFlavor with valid client and compute order, the plugin
+	// should throw a NoAvailableResourcesException when no appropriate flavor fits the order
+	// hardware requirements
 	@Test
-	public void testConvertToIntegerSuccessfully() {
+	public void testFindSmallestFlavorFail() throws UnexpectedException {
 		// set up
-		String number = TEMPLATE_CPU_VALUE;
-		int expected = CPU_VALUE_2;
+		Mockito.doReturn(null).when(this.plugin).getBestFlavor(
+				Mockito.any(Client.class), Mockito.any(ComputeOrder.class));
 
 		// exercise
-		int value = this.plugin.convertToInteger(number);
+		try {
+			this.plugin.findSmallestFlavor(this.client, this.computeOrder);
+			Assert.fail();
+		} catch (NoAvailableResourcesException e) {
+		    Assert.assertEquals("No available resources.", e.getMessage());
```

Comment:
The same suggestion above defines this string in a constant.

Reply:
this seems to be used only once in the entire test class, but ok.